### PR TITLE
Azure Policy Policy Effect + Terraform Scenarios

### PR DESCRIPTION
**Pull request by liamjvs**
## Overview/Summary

Expanded the section on Azure Policy effects with Terraform.

## This PR fixes/adds/changes/removes

1. Expanded the section around policy effects when using Azure Policy and Terraform

### Breaking Changes

None.

## Testing Evidence

Addition to documentation.

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/pulls)
- [ ] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [ ] Updated the ["RELEASE"](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues) page with details needed for when this code change is released.

### CODE REVIEW

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]

File: docs/wiki/[User-Guide]-Getting-Started.md [hunk 1]
```diff
@@ -53,19 +53,38 @@ However, when application teams choose to deploy resources with Terraform, the f
...
+    - The application team performs an additional Terraform run where Terraform discovers that the resource has drifted away from the Terraform code and state. Terraform will try to correct the resource by either changing the property back or re-creating it.
+    - Azure Policy will remediate the resource so that it is compliant with the guardrails of the platform.
+
+    An example of this can be enforcing soft-delete on Key Vaults or enforcing Transport Data Encryption (TDE) through `Append`/`Modify` policies; the properties will not be defined in Terraform but will be remediated via Azure Policy resulting in the above loop. This is almost always problematic when managing the resource with Terraform, however there is a rare case where the modified property of the resource is not tracked in Terraform state, then there will be no issue.
+
+    An exception to the above is when the use of `DeployIfNotExists` does not modify the in-scope resource of the Terraform deployment but instead deploys a child resource to the non-compliant resource:
```
Comment (jtracey93) [inline docs/wiki/[User-Guide]-Getting-Started.md:79]:
```suggestion
    An exception to the above is when the use of `DeployIfNotExists` does not modify the in-scope resource of the Terraform deployment but instead deploys a child or extension resource to the non-compliant resource:
```

File: docs/wiki/[User-Guide]-Getting-Started.md [hunk 2]
```diff
@@ -53,19 +53,38 @@ However, when application teams choose to deploy resources with Terraform, the f
...
+    - A resource is deployed by the application team using Terraform.
+    - Azure Policy deploys a child resource to the resource to ensure the resource is compliant.
```
Comment (jtracey93) [inline docs/wiki/[User-Guide]-Getting-Started.md:81]:
```suggestion
    - Azure Policy deploys a child or extension resource to the resource to ensure the resource is compliant.
```

File: docs/wiki/[User-Guide]-Getting-Started.md [hunk 3]
```diff
@@ -53,19 +53,38 @@ However, when application teams choose to deploy resources with Terraform, the f
...
+    - An additional Terraform run is performed, and there is no state-drift as Terraform does not need to modify or alter the child resource.
+    
+    An example of this is deploying Diganostic Settings to a resource, a Private DNS Zone Group to a Private Endpoint or a security rule to a Network Security Group.
```
Comment (jtracey93) [inline docs/wiki/[User-Guide]-Getting-Started.md:84]: Not sure on NSG rules here as they can be deployed in a NSG resource on the properties.securityRules array

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: docs/wiki/[User-Guide]-Getting-Started.md (hunk 3 above)
Comment (liamjvs) [reply docs/wiki/[User-Guide]-Getting-Started.md]: Agree - it was in reference to the Append section which was removed. Removed the NSG reference now too.

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**